operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -53,15 +55,57 @@ where
     )
     .await?;
 
-    let bytes = gdal::vsi::get_vsi_mem_file_bytes_owned(file_path)?;
-
-    Ok(bytes)
+    if timesteps.len() == 1 {
+        Ok(timesteps
+            .pop()
+            .expect("there should be exactly one timestep"))
+    } else {
+        Err(Error::InvalidNumberOfTimeSteps {
+            expected: 1,
+            found: timesteps.len(),
+        })
+    }
 }
 
-#[derive(Debug, Clone, Copy)]
-pub struct GdalGeoTiffDatasetMetadata {
-    pub no_data_value: Option<f64>,
-    pub spatial_reference: SpatialReference,
+#[allow(clippy::too_many_arguments)]
+pub async fn raster_stream_to_geotiff_bytes<T, C: QueryContext + 'static>(
+    processor: Box<dyn RasterQueryProcessor<RasterType = T>>,
+    query_rect: RasterQueryRectangle,
+    query_ctx: C,
+    gdal_tiff_metadata: GdalGeoTiffDatasetMetadata,
+    gdal_tiff_options: GdalGeoTiffOptions,
+    tile_limit: Option<usize>,
+    conn_closed: BoxFuture<'_, ()>,
+    tiling_specification: TilingSpecification,
+) -> Result<Vec<Vec<u8>>>
+where
+    T: Pixel + GdalType,
+{
+    let file_path = PathBuf::from(format!("/vsimem/{}/", uuid::Uuid::new_v4()));
+
+    let result = raster_stream_to_geotiff(
+        &file_path,
+        processor,
+        query_rect,
+        query_ctx,
+        gdal_tiff_metadata,
+        gdal_tiff_options,
+        tile_limit,
+        conn_closed,
+        tiling_specification,
+    )
+    .await?
+    .into_iter()
+    .map(|x| {
+        gdal::vsi::get_vsi_mem_file_bytes_owned(
+            x.params
+                .expect("params is always set in raster_stream_to_geotiff")
```

Comment:
be26049d69aa5e5b30df269083c554ad1763b180

I don't believe this can occur at the moment, but for now I decided to use an empty vector to represent a gap. There are other possibilities (skip gaps, create an empty tiff and read that in, ...), but I did not get a definitive answer in Slack, so I took the empty vector route.